staging/src/k8s.io/apiserver/pkg/util/shufflesharding/shufflesharding_test.go

```diff
@@ -279,26 +282,43 @@ func BenchmarkShuffleAndDealToSlice(b *testing.B) {
 }
 
 func TestUniformDistribution(t *testing.T) {
-	deckSize, handSize := int32(128), int32(3)
-	handCoordinateMap := make(map[int]int)
+	var aHand [3]int
+	deckSize := int32(128)
+	handCoordinateMap := make(map[int]int) // maps coded hand to count of times seen
 
-	allCoordinateCount := 128 * 127 * 126 / 6
+	fallingFactorial := 128 * 127 * 126
+	allCoordinateCount := fallingFactorial / 6
+	hashMax := uint64(16 << uint32(math.Ceil(math.Log2(float64(fallingFactorial)))))
```

Comment:
Could we have `hashMax := fallingFactorial << (64 - maxHashBits)` here? The center will be 2^(64-maxHashBits). Seems easy to be understood.

Reply:
It is easy to prove that whenever hashMax is an integer multiple of `fallingFactorial` the distribution of hand occurrences is perfect --- every hand gets dealt the same number of times.  A more interesting test is when hashMax is a power of 2 rather than a multiple of `fallingFactorial`.

We could replace `16 << X` with `1 << (X + 64 - maxHashBits)`.